ssvsigner/web3signer/types.go

```diff
@@ -0,0 +1,124 @@
+}
+
+type KeyManagerResponseData struct {
+	Status  Status `json:"status"`
+	Message string `json:"message"`
+}
+
+type Status string
+
+const (
+	StatusImported   Status = "imported"
+	StatusDuplicated Status = "duplicate"
+	StatusDeleted    Status = "deleted"
+	StatusNotActive  Status = "not_active"
+	StatusNotFound   Status = "not_found"
+	StatusError      Status = "error"
+)
+
+type SignRequest struct {
+	ForkInfo                    ForkInfo                              `json:"fork_info"`
+	SigningRoot                 string                                `json:"signing_root,omitempty"`
+	Type                        SignedObjectType                      `json:"type"`
+	Attestation                 *phase0.AttestationData               `json:"attestation,omitempty"`
+	BeaconBlock                 *BeaconBlockData                      `json:"beacon_block,omitempty"`
+	VoluntaryExit               *phase0.VoluntaryExit                 `json:"voluntary_exit,omitempty"`
+	AggregateAndProof           *AggregateAndProofData                `json:"aggregate_and_proof,omitempty"`
+	AggregationSlot             *AggregationSlotData                  `json:"aggregation_slot,omitempty"`
+	RandaoReveal                *RandaoRevealData                     `json:"randao_reveal,omitempty"`
+	SyncCommitteeMessage        *SyncCommitteeMessageData             `json:"sync_committee_message,omitempty"`
+	SyncAggregatorSelectionData *SyncCommitteeAggregatorSelectionData `json:"sync_aggregator_selection_data,omitempty"`
+	ContributionAndProof        *altair.ContributionAndProof          `json:"contribution_and_proof,omitempty"`
+	ValidatorRegistration       *v1.ValidatorRegistration             `json:"validator_registration,omitempty"`
+}
+
+type ForkInfo struct {
+	Fork                  *phase0.Fork `json:"fork"`
+	GenesisValidatorsRoot phase0.Root  `json:"genesis_validators_root"`
+}
+
+type SignedObjectType string
+
+const (
+	AggregationSlot                   SignedObjectType = "AGGREGATION_SLOT"
+	AggregateAndProof                 SignedObjectType = "AGGREGATE_AND_PROOF"
+	Attestation                       SignedObjectType = "ATTESTATION"
+	Block                             SignedObjectType = "BLOCK" // TODO: not tested
+	BlockV2                           SignedObjectType = "BLOCK_V2"
+	Deposit                           SignedObjectType = "DEPOSIT" // TODO: not tested
+	RandaoReveal                      SignedObjectType = "RANDAO_REVEAL"
+	VoluntaryExit                     SignedObjectType = "VOLUNTARY_EXIT" // TODO: not tested
+	SyncCommitteeMessage              SignedObjectType = "SYNC_COMMITTEE_MESSAGE"
+	SyncCommitteeSelectionProof       SignedObjectType = "SYNC_COMMITTEE_SELECTION_PROOF"
+	SyncCommitteeContributionAndProof SignedObjectType = "SYNC_COMMITTEE_CONTRIBUTION_AND_PROOF"
+	ValidatorRegistration             SignedObjectType = "VALIDATOR_REGISTRATION"
+)
+
+type BeaconBlockData struct {
+	Version     string                    `json:"version"`
```

Comment:
lets use our custom DataVersion wrapper instead of string (see https://github.com/ssvlabs/ssv/pull/2028#discussion_r1986295637)

Reply:
done